### ISSUE TITLE
Can no close the app on full-screen mode

### ISSUE THREAD

**Issue by muhghazaliakbar** [gitbutlerapp/gitbutler#5869]
### Version

0.14.4

### Operating System

macOS

### Distribution Method

dmg (Apple Silicon)

### Describe the issue

The application can not be closed when you are in full-screen mode. Here's the attached video about the issue

https://github.com/user-attachments/assets/9ca2e152-d7d9-4757-b3d3-edfd85f90358

### How to reproduce

Just go to full-screen mode and try to close the app by clicking the red button on the top left

**Comment by Byron** [comment]
Thanks a lot for reporting!

This issue was easily reproducible as well. On top of that, I noticed that after clicking the close button, to focus seems to shift onto another application, so there seems to be *some* side-effect.

I'd think that this related to the special behaviour on MacOS that only closes the WIndow instead of shutting down the application. Window-closing doesn't seem to work when maximised.

**Comment by muhghazaliakbar** [comment]
> Thanks a lot for reporting!
> 
> This issue was easily reproducible as well. On top of that, I noticed that after clicking the close button, to focus seems to shift onto another application, so there seems to be _some_ side-effect.
> 
> I'd think that this related to the special behaviour on MacOS that only closes the WIndow instead of shutting down the application. Window-closing doesn't seem to work when maximised.

You're right! it doesn't matter the application is closed or just minimized by MacOS, but it should be working, right?

**Comment by PavelLaptev** [comment]
Sorry, @Byron, I removed the "UI" tag. To me, this is not a UI issue. Traffic lights are a native thing, visually not broken, but they don’t work.

**Comment by Byron** [comment]
@PavelLaptev Apologies, I will keep that in mind.

> It doesn't matter the application is closed or just minimized by MacOS, but it should be working, right?

Definitely! I merely dumped everything I know about it as to help others or myself in future.

**Comment by Caleb-T-Owens** [comment]
I've noticed this before. I'm going to add it to the end of my TODO list as in my mind this is not a high priority. Anyone else can feel free to take a look however.
